src/cpl.rs

```diff
@@ -324,30 +358,46 @@ impl FromStr for CslStringList {
     }
 }
 
-/// Convenience for creating a [`CslStringList`] from a slice of _key_/_value_ tuples.
-///
-/// # Example
-///
-/// ```rust, no_run
-/// use gdal::cpl::CslStringList;
-///
-/// let opts = CslStringList::try_from(&[("One", "1"), ("Two", "2"), ("Three", "3")]).expect("known valid");
-/// assert_eq!(opts.len(), 3);
-/// ```
-impl<const N: usize> TryFrom<&[(&str, &str); N]> for CslStringList {
-    type Error = GdalError;
-
-    fn try_from(pairs: &[(&str, &str); N]) -> Result<Self> {
+impl FromIterator<CslStringListEntry> for CslStringList {
+    fn from_iter<T: IntoIterator<Item = CslStringListEntry>>(iter: T) -> Self {
         let mut result = Self::default();
-        for (k, v) in pairs {
-            result.set_name_value(k, v)?;
+        for e in iter {
+            result.add_entry(&e).unwrap_or_default()
+        }
+        result
+    }
+}
+
+impl<'a> FromIterator<&'a str> for CslStringList {
+    fn from_iter<T: IntoIterator<Item = &'a str>>(iter: T) -> Self {
+        iter.into_iter()
+            .map(Into::<CslStringListEntry>::into)
+            .collect()
+    }
+}
+
+impl FromIterator<String> for CslStringList {
+    fn from_iter<T: IntoIterator<Item = String>>(iter: T) -> Self {
+        iter.into_iter()
+            .map(Into::<CslStringListEntry>::into)
+            .collect()
+    }
+}
+
+impl Extend<CslStringListEntry> for CslStringList {
+    fn extend<T: IntoIterator<Item = CslStringListEntry>>(&mut self, iter: T) {
+        for e in iter {
+            self.add_entry(&e).unwrap_or_default();
         }
-        Ok(result)
     }
 }
 
-/// Represents an entry in a [`CslStringList`], which is ether a single token (`Flag`),
-/// or a `name=value` assignment (`Pair`).
+/// Represents an entry in a [`CslStringList`]
+///
+/// An Entry is ether a single token (`Flag`), or a `name=value` assignment (`Pair`).
+///
+/// Note: When constructed directly, assumes string values do not contain newline characters nor
```

Comment:
We should probably spell out more clearly that those values will be ignored. And not just here, but also on `CslStringList`.

Reply:
`CslStringList` actually propagates the error.
https://github.com/georust/gdal/blob/16f2aefd16e7d01cf27e2fce680e7eed93a7a5f2/src/cpl.rs#L116-L117